pkg/k8sapi/k8sutils.go

```diff
@@ -45,9 +45,7 @@ func getIPAMDCacheFilters() map[client.Object]cache.ByObject {
 			&corev1.Node{}: {
 				Field: fields.Set{"metadata.name": nodeName}.AsSelector(),
 			},
-			&rcscheme.CNINode{}: {
-				Field: fields.Set{"metadata.name": nodeName}.AsSelector(),
-			},
+			&eniconfigscheme.ENIConfig{}: {},
```

Comment:
Why do we change this?

Reply:
the CNINode cache is redundant, it's due to the merge history on this release-branch... we will only cache CNINode when SGP is enabled.
ENIConfig is currently missing the cache, we should also cache it, right

Reply:
Got it. I think ENIConfig should be fine since we have no way to reduce the scope as other types.